File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;
class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> stack = new Stack<>();
        for(int move : moves){
            for(int i=0; i<board.length; i++){
                if(board[i][move-1] != 0){
                    if(!stack.isEmpty() && stack.peek() == board[i][move-1]){
                        stack.pop();
                        answer+=2;
                    }else{
                        stack.push(board[i][move-1]);
                    }
                     //다음 인형을 찾을 수 있도록 기존 인형을 제거
                    board[i][move-1] = 0;
                    break;
                }
            }
        }
        return answer;
        
        }
    }

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0;
    const basket = [];
    for(let i=0; i<moves.length; i++){
        for(let j=0; j<board.length; j++){
            let choice = board[j][moves[i]-1];
            if(choice !== 0){
                if(basket[basket.length-1] == choice){
                    basket.pop();
                    board[j][moves[i]-1] = 0;
                    answer+=2;
                }else{
                    basket.push(choice);
                    board[j][moves[i]-1] = 0;
                }
                break;
            }
        }
    }
    return answer;
}